src/main/java/org/brapi/test/BrAPITestServer/service/germ/GermplasmService.java

```diff
@@ -108,8 +109,58 @@ public List<Germplasm> findGermplasm(@Valid GermplasmSearchRequest request, Meta
 		return germplasms;
 	}
 
-	public Page<GermplasmEntity> findGermplasmEntities(@Valid GermplasmSearchRequest request, Metadata metadata) {
+	public List<Germplasm> findGermplasmWithoutPaging(@Valid GermplasmSearchRequest request) {
+		List<GermplasmEntity> entities = findGermplasmEntitiesWithoutPaging(request);
+		return entities.stream().map(this::convertFromEntity).collect(Collectors.toList());
+	}
+
+	public Page<GermplasmEntity> findGermplasmEntities(@Valid GermplasmSearchRequest request, Metadata metadata)
+		throws BrAPIServerException {
 		Pageable pageReq = PagingUtility.getPageRequest(metadata);
+
+		SearchQueryBuilder<GermplasmEntity> searchQuery = buildGermplasmSearchQuery(request);
+
+		// First run the query without the fetching all the entities, grabbing the IDs only and paginating and sorting them.
+		Page<GermplasmEntity> germs = germplasmRepository.findAllBySearchPaginatingWithFetches(searchQuery, pageReq);
```

Comment:
```suggestion
		// First query germplasm with pagination and sorting but without fetching all the related entities.
		Page<GermplasmEntity> germs = germplasmRepository.findAllBySearchPaginatingWithFetches(searchQuery, pageReq);
```

The comment confused me, It made it sound like the returned `GermplasmEntity` might only have the `id` field set on them. I've suggested one possible change.

Reply:
This comment is actually not needed anymore, because now the logic it was trying to enumerate is self-contained within `findAllBySearchPaginatingWithFetches`.  I'm replacing it with a comment that illustrates the reason why we need to use paginate with fetches in this scenario rather than exactly what the call is doing, which can be found by tunneling into the method itself.